### DISCUSSION TITLE
How to make a custom dependency visible on PATH during install?

### DISCUSSION THREAD

**Discussion by danirukun** [Homebrew/discussions#2893] [Writing Formulae/Casks] [answered]
I am writing a formula for an app that requires a specific version of a binary, so I have to download it using cURL.

However, in the `Makefile`, it tries looking for `rebar3`, but it doesn't work, because the current directory is not on `PATH`, so instead the `./rebar3` syntax is needed. What is the best approach for these situations?

```ruby
class ErlangLs < Formula
  desc "The Erlang Language Server"
  homepage "https://erlang-ls.github.io/"
  url "https://github.com/erlang-ls/erlang_ls/archive/refs/tags/0.21.2.tar.gz"
  sha256 "8a0466f525091a483401a958c6c1f85cd3203c6103b0fbabf784be449031b42c"
  license "Apache-2.0"

  depends_on "erlang@23"

  def install
    system "curl", "https://github.com/erlang/rebar3/releases/download/3.11.0/rebar3", "--output", "rebar3"
    chmod "+x", "./rebar3"
    system "alias", "rebar3=./rebar3", "make"
    system "make", "install"
  end

  test do
    system "false"
  end
end

```

**Comment by alebcay** [comment, marked as answer]
You're on the right track. Here's how I would write it.

```ruby
class ErlangLs < Formula
  desc "The Erlang Language Server"
  homepage "https://erlang-ls.github.io/"
  url "https://github.com/erlang-ls/erlang_ls/archive/refs/tags/0.21.2.tar.gz"
  sha256 "8a0466f525091a483401a958c6c1f85cd3203c6103b0fbabf784be449031b42c"
  license "Apache-2.0"

  depends_on "erlang@23"

  resource "rebar3" do
    url "https://github.com/erlang/rebar3/releases/download/3.11.0/rebar3"
    sha256 "407286134cc512c433d371e54456fe4c87936174d0d141301ee81fb9e6107683"
  end

  def install
    # Download the rebar3 binary to the directory where the build is happening
    resource("rebar3").stage buildpath

    # Make it executable
    chmod "+x", buildpath/"rebar3"

    # Append the build directory to the PATH environment variable
    ENV.append_path "PATH", buildpath

    system "make"
    system "make", "install"
  end

  test do
    system "false"
  end
end
```

(Fair warning, I haven't tried running/testing this sample at all.)

Instead of `curl`, consider using a Homebrew `resource` for the `rebar3` binary. This allows Homebrew to cache and checksum anything extra downloaded during the build.

You can modify environment variables with the `ENV.append_path`, `ENV.prepend_path`, etc. functions. They do what you'd expect - append or prepend a particular environment variable with the given argument (colon-separated paths). It's used for modifying the `PATH` environment variable (e.g. [in the `dockerize` formula](https://github.com/Homebrew/homebrew-core/blob/a0edf7f9406615c2f0ffcba08c0614ba8613c816/Formula/dockerize.rb#L25)), and some other similarly formatted environment variables like [`PYTHONPATH`](https://github.com/Homebrew/homebrew-core/blob/a0edf7f9406615c2f0ffcba08c0614ba8613c816/Formula/easy-tag.rb#L35), [`PKG_CONFIG_PATH`](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/freeswitch.rb#L134), and [`CRYSTAL_LIBRARY_PATH`](https://github.com/Homebrew/homebrew-core/blob/a0edf7f9406615c2f0ffcba08c0614ba8613c816/Formula/crystal.rb#L78).

**Reply by danirukun** [reply]
Thanks, this was exactly what I was looking for!
